modules/context/repo.go

```diff
@@ -177,54 +177,72 @@ func RepoAssignment() macaron.Handler {
 	return func(ctx *Context) {
 		var (
 			owner *models.User
+			repo  *models.Repository
```

Comment:
Why are these changes to `RepoAssignment()` necessary? It seems to me that adding the `context.LoadRepoUnits()` and `context.CheckUnit(models.UnitTypeCode)` handlers should be sufficient (along with the changes to `repo.CanBeForked()`)

Reply:
I think it needs load repo because `context.LoadRepoUnits()` needs `ctx.Reop` to context but fork's route only has `reopid` no repopath.

Reply:
Good catch, you're right.

But since we only need to load `ctx.Repo`, maybe it would be better to add a separate handler for loading `:repoid`, instead of making `RepoAssignment()` more complicated by adding a big `if`/`else`?

Reply:
@ethantkoenig I agree with you to make a new middleware to do that but not change `ReopAssignment`